### PR TITLE
Add cluster specific configuration section

### LINKED ISSUE #373: Move GKE private cluster documentation to Installation page
PR #371 adds some useful information for using GKE private clusters, and this information belongs in the GKE section of the installation docs.

### PR DESCRIPTION
**Pull request by grampelberg**
Fixes #373.

### CODE REVIEW

File: linkerd.io/content/2/getting-started/_index.md [hunk 1]
```diff
@@ -45,6 +45,14 @@ kubectl version --short
 In the next step, we will install the CLI and validate that your cluster is
 ready to install the control plane.
 
+### GKE
+
+If installing Linkerd on GKE, there are some extra steps required depending on
+how your cluster has been configured. If you are using any of these features,
+check out the additional instructions.
+
+- [Private clusters](/2/reference/cluster-configuration/#private-clusters)
```
Comment (klingerf) [inline linkerd.io/content/2/getting-started/_index.md]: Should this list also include RBAC, linked to `/2/reference/cluster-configuration/#rbac`?

File: linkerd.io/content/2/tasks/install.md [hunk 2]
```diff
@@ -35,6 +35,16 @@ Linkerd by running
 linkerd check --pre
 ```
 
+### GKE
+
+If installing Linkerd on GKE, there are some extra steps required depending on
+how your cluster has been configured. If you are using any of these features,
+check out the additional instructions.
+
+- [Private clusters](/2/reference/cluster-configuration/#private-clusters)
```
Comment (klingerf) [inline linkerd.io/content/2/tasks/install.md]: Same comment here as above.

File: linkerd.io/content/2/getting-started/_index.md (hunk 1 above)
Comment (wmorgan) [inline linkerd.io/content/2/getting-started/_index.md]: Do we want to add an "RBAC enabled" link?

Comment (grampelberg) [reply linkerd.io/content/2/getting-started/_index.md]: I went and did some testing, it isn't required for modern GKE clusters anymore (1.12+). There looks like a bug in `check` that once fixed will get folks looking at the troubleshooting docs right away and see the RBAC section. I can add it, but it felt like we were getting to the point where it wasn't required anywhere outside the check docs yet.

**Pull request merged**